app/src/main/java/org/covidwatch/android/presentation/test/DatePickerMixin.kt

```diff
@@ -0,0 +1,28 @@
+package org.covidwatch.android.presentation.test
+
+import android.app.DatePickerDialog
+import androidx.fragment.app.Fragment
+import java.util.*
+
+interface DatePickerMixin {
+    val testQuestionsViewModel: TestQuestionsViewModel
+    fun showDatePicker(fragment: Fragment) {
+        val calendar = Calendar.getInstance()
+        val dialog = DatePickerDialog(
+            fragment.requireContext(),
+            DatePickerDialog.OnDateSetListener { _, year, monthOfYear, dayOfMonth ->
+                calendar.set(Calendar.YEAR, year)
+                calendar.set(Calendar.MONTH, monthOfYear)
+                calendar.set(Calendar.DAY_OF_MONTH, dayOfMonth)
+
+                testQuestionsViewModel.onDateSelected(calendar.time)
+            },
+            calendar.get(Calendar.YEAR),
+            calendar.get(Calendar.MONTH),
+            calendar.get(Calendar.DAY_OF_MONTH)
+        )
+        dialog.datePicker.maxDate = Date().time
+
+        dialog.show()
+    }
+}
```

Comment:
Let's make an [extensions function](https://kotlinlang.org/docs/reference/extensions.html) with callback so we don't depend of `TestQuestionsViewModel`?

`./presentation/util/FragmentExtensions.kt`
```kotlin
fun Fragment.showDatePicker(callback: (date: Date) -> Unit) {
    val calendar = Calendar.getInstance()
    val dialog = DatePickerDialog(
        requireContext(),
        DatePickerDialog.OnDateSetListener { _, year, monthOfYear, dayOfMonth ->
            calendar.set(Calendar.YEAR, year)
            calendar.set(Calendar.MONTH, monthOfYear)
            calendar.set(Calendar.DAY_OF_MONTH, dayOfMonth)

            callback(calendar.time)
        },
        calendar.get(Calendar.YEAR),
        calendar.get(Calendar.MONTH),
        calendar.get(Calendar.DAY_OF_MONTH)
    )
    dialog.datePicker.maxDate = Date().time

    dialog.show()
}
```

Reply:
Done.